core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -61,7 +61,16 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   var pyFiles: String = null
   var isR: Boolean = false
   var action: SparkSubmitAction = null
-  val sparkProperties: HashMap[String, String] = new HashMap[String, String]()
+  val sparkProperties: HashMap[String, String] = new HashMap[String, String]() {
+    override def put(k: String, v: String): Option[String] = {
```

Comment:
Agree with Sean.  The behavior you're introducing that always ignores non-spark properties seems right to me, but extending HashMap here seems a little weird.